### PR TITLE
added CONTRIBUTING.md and updated tox ini to align with travis

### PR DESCRIPTION
**Pull request by chrisbeardy**
I added this in as I think it will help people like me to contribute, mainly to remember to run black!

Feel free to add anymore configurations to ignore for black.

### CODE REVIEW

**Comment by chrisbeardy** [comment]
it could be worth looking at using [pre-commit](https://pre-commit.com/) too for this.

File: tox.ini [hunk 1]
```diff
@@ -4,7 +4,7 @@
 # and then run "tox" from this directory.
 
 [tox]
-envlist = py27, py37
+envlist = py27, py35, py36
```
Comment (stlehmann) [inline tox.ini]: Including py35 and py36 make sense but why did you remove py37 from the list?

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+[tool.black]
```
Comment (stlehmann) [inline pyproject.toml]: What is this .toml file for? I don't know the format ;) Is it just for black formatting?

Comment (chrisbeardy) [reply pyproject.toml]: Toml is just another markup language, the pyproject. toml file is a project file that some plugins like black use for configuration, in some cases I've seen it used with setup.py too. I've not used black before, normally used autopep8 but I quite like black! Reading the docs it looks for pyproject.toml for configuration.

File: tox.ini (hunk 1 above)
Comment (chrisbeardy) [reply tox.ini]: To align it with the Travis build. As Travis only tests 27, 35 and 36 I thought it makes sense for you tox to use the same? Could add 37 back?

**Comment by chrisbeardy** [comment]
I mainly copy and pasted the contributing from other projects as most are the same really, I also find the deluge bittorrent client a really good repo to steal ideas from....

**Comment by chrisbeardy** [comment]
And when I say steal I mean I really like the way they have there project set up and the tools they use. I think precommit is a really good tool.

File: tox.ini (hunk 1 above)
Comment (stlehmann) [reply tox.ini]: Ah OK, I didn' t have in mind that we only test with 3.6 in Travis. I think that was due to Python3.7 not being available in Travis at that time. We should change that. Means adding py37 test to tox and Travis would be the move here, I think.

Comment (chrisbeardy) [reply tox.ini]: agreed, ill do that
